homeassistant/components/dynalite/config_flow.py

```diff
@@ -18,19 +20,61 @@ def __init__(self) -> None:
         """Initialize the Dynalite flow."""
         self.host = None
 
+    def existing_host_entry(self, config) -> Any:
+        """Check if a specific host is already configured and return its entry or None."""
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            if entry.data[CONF_HOST] == config[CONF_HOST]:
+                return entry
+        return None
+
+    async def check_bridge_connection(self, config) -> bool:
+        """Check whether this bridge is available to connect."""
+        bridge = DynaliteBridge(self.hass, config)
+        if not await bridge.async_setup():
+            LOGGER.error("Unable to setup bridge - config=%s", config)
+            return False
+        return True
+
     async def async_step_import(self, import_info: Dict[str, Any]) -> Any:
         """Import a new bridge as a config entry."""
         LOGGER.debug("Starting async_step_import - %s", import_info)
         host = import_info[CONF_HOST]
-        for entry in self.hass.config_entries.async_entries(DOMAIN):
-            if entry.data[CONF_HOST] == host:
-                if entry.data != import_info:
-                    self.hass.config_entries.async_update_entry(entry, data=import_info)
-                return self.async_abort(reason="already_configured")
+        entry = self.existing_host_entry(import_info)
+        if entry:
+            if entry.data != import_info:
+                self.hass.config_entries.async_update_entry(entry, data=import_info)
+            return self.async_abort(reason="already_configured")
         # New entry
-        bridge = DynaliteBridge(self.hass, import_info)
-        if not await bridge.async_setup():
+        if not await self.check_bridge_connection(import_info):
             LOGGER.error("Unable to setup bridge - import info=%s", import_info)
             return self.async_abort(reason="no_connection")
         LOGGER.debug("Creating entry for the bridge - %s", import_info)
         return self.async_create_entry(title=host, data=import_info)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        # This is for backwards compatibility.
+        return await self.async_step_init(user_input)
+
+    async def async_step_init(self, user_input=None):
+        """Handle a flow start."""
+        if user_input is not None:
+            # New entry
+            if self.existing_host_entry(user_input):
+                return self.async_abort(reason="already_configured")
+            if not await self.check_bridge_connection(user_input):
+                LOGGER.error("Unable to setup bridge - init user_input=%s", user_input)
+                return self.async_abort(reason="no_connection")
```

Comment:
You should probably show the form again with the error information instead of aborting. More user friendly in case of typos, etc...

Reply:
thanks. is there a way to tell the user why i am showing the form again (like a red "connection problem" on the top)?

Reply:
Yes, you can pass the `errors` parameter to `async_show_form`. Take a look at the Volumio config flow for example.

Reply:
ok. will work on it.

Reply:
ok. just pushed the change.
do you know why my strings are still showing as [%key:common::config_flow::data::host%] in the UI? I ran the script.translation on the component to auto-generate, but it still shows this way. The documentation wasn't 100% clear - is this something that gets changed in the back end after the PR is pushed?

Reply:
Yeah, that's normal when using the common strings. It's just like that in your local environment though, don't worry about it

Reply:
thanks. is there a way to tell it to convert the common strings so when i test it locally, i will see the real string?

Reply:
Not AFAIK. Probably something that will be fixed in the tooling at some point.

Reply:
ok. thanks